SmartDeviceLink/private/SDLIAPDataSession.m

```diff
@@ -58,10 +58,24 @@ - (void)writeDataToSessionStream {
             if (bytesWritten >= 0) {
                 if (bytesWritten == bytesRemaining) {
                     [self.sendDataQueue popBuffer];
-                } else if (bytesRemaining > bytesWritten) {
+                } else if (bytesRemaining > bytesWritten && remainder.length > bytesWritten) {
                     // Cleave the sent bytes from the data, the remainder will sit at the head of the queue
-                    SDLLogV(@"SDLIAPDataSession writeDataToSessionStream bytes written %ld", (long)bytesWritten);
-                    [remainder replaceBytesInRange:NSMakeRange(0, (NSUInteger)bytesWritten) withBytes:NULL length:0];
+                    @try {
+                        SDLLogV(@"SDLIAPDataSession writeDataToSessionStream bytes written %ld", (long)bytesWritten);
+                        NSData *rem = [remainder subdataWithRange:NSMakeRange((NSUInteger)bytesWritten, remainder.length - 1)];
```

Comment:
```suggestion
                        NSData *rem = [remainder subdataWithRange:NSMakeRange((NSUInteger)bytesWritten, remainder.length - (NSUInteger)bytesWritten)];
```
The second parameter for `NSMakeRange` is the length of the range, not its position. Right now it will fail the initial block because it tries creates a range that exceeds the length of the data and thus will force it into the `replaceBytesInRange` block.